src/pydiverse/pipedag/materialize/debug.py

```
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import random
+
+import structlog
+
+from pydiverse.pipedag import ConfigContext, Table
+from pydiverse.pipedag.backend import SQLTableStore
+from pydiverse.pipedag.backend.table.sql.ddl import DropTable
+from pydiverse.pipedag.context import TaskContext
+from pydiverse.pipedag.materialize.core import MaterializingTask
+from pydiverse.pipedag.util.hashing import stable_hash
+
+
+def materialize_table(
+    table: Table,
+    debug_suffix: str | None = None,
+    flag_task_debug_tainted: bool = True,
+    keep_table_name: bool = True,
+    replace_table: bool = True,
```

The naming of `replace_table` might be optimized. Some dataframe materializations use if_exists="replace". This is a crude enum technique. However, people might be familiar with it. And we could raise an exception for all other string values.
Another alternative would be to rename it `drop_if_exists`. This would be closest to what this feature actually does and thus makes it easier to understand error messages reported by the database backend.